api/data_explorer/dataset_faceted_search.py

```diff
@@ -0,0 +1,33 @@
+"""Subclass of FacetedSearch for Data explorer datasets."""
+
+import json
+
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import FacetedSearch
+from elasticsearch_dsl import TermsFacet
+
+
+def _get_index_name():
+    """Gets index name from /app/dataset.json."""
+    # JSON doesn't support comments
+    # (https://plus.google.com/+DouglasCrockfordEsq/posts/RK8qyGVaGSr). Remove
+    # comments before parsing.
+    with open('/app/dataset.json') as f:
+        dataset = json.loads('\n'.join([row for row in f.readlines() if len(row.split('//')) == 1]))
+        return dataset['name']
+
+
+class DatasetFacetedSearch(FacetedSearch):
+    index = _get_index_name()
+    using = Elasticsearch('elasticsearch:9200')
```

Comment:
Seems like something we shouldn't be hard coding, as the elasticsearch server won't always be local. Can we extract this?